cache-filter/src/filter/http.rs

```diff
@@ -367,26 +375,42 @@ impl CacheFilter {
             upstream: upstream_builder.build(&cluster_name, timeout),
         })
     }
+
+    #[cfg(feature = "visible_logs")]
+    #[allow(dead_code)]
+    fn on_http_response_header(&mut self, _: usize) -> Action {
+        let (key, val) = crate::log::visible_logs::get_logs_header_pair(self.context_id);
+        self.add_http_response_header(key.as_ref(), val.as_ref());
+        Action::Continue
+    }
 }
 
 impl Context for CacheFilter {
     fn on_http_call_response(&mut self, token_id: u32, _: usize, body_size: usize, _: usize) {
         info!(
-            "ctxt {}: received response from 3scale: token: {}",
-            self.context_id, token_id
+            self.context_id,
+            "received response from 3scale: token: {}", token_id
         );
         match self.get_http_call_response_body(0, body_size) {
             Some(bytes) => {
                 match Authorization::from_str(std::str::from_utf8(&bytes).unwrap()) {
                     Ok(Authorization::Status(response)) => {
                         if response.is_authorized() || response.usage_reports().is_some() {
                             if let Err(e) = self.handle_auth_response(&response) {
-                                warn!(
-                                    "ctxt {}: handling auth response failed: {:?}",
-                                    self.context_id, e
-                                );
+                                warn!(self.context_id, "handling auth response failed: {:?}", e);
                                 request_process_failure(self, self)
                             }
+                        } else if cfg!(feature = "visible_logs") {
+                            #[cfg(feature = "visible_logs")]
+                            {
+                                let (key, val) =
+                                    crate::log::visible_logs::get_logs_header_pair(self.context_id);
+                                self.send_http_response(
+                                    403,
+                                    vec![(key.as_ref(), val.as_ref())],
+                                    Some(response.reason().unwrap().as_bytes()),
+                                );
+                            }
```

Comment:
Same as above, you only really need to do your thing with the logs and use a specific body (thus reducing duplication).